### PR TITLE
Feature/proj 164

### PR DESCRIPTION
**Pull request by Lelik7777**
### Overview
This pull request introduces several features, updates, and refactoring efforts across the codebase. The main focus is on enhancing account management functionality, improving modal handling, and optimizing the sidebar component.

### Features Added
1. **Dynamic Base URL in `useAccountManagement`**:
   - Implemented logic to retrieve a dynamic base URL for API calls within the `useAccountManagement` hook.

2. **Loader Component**:
   - Created a new `Preloader.ts` component with styles in `shared/ui` to provide a consistent loading experience across the application.

3. **Settings Tabs Hook**:
   - Introduced a new hook `useSettingsTabs.tsx` for managing settings tab states.

4. **Modal Hook**:
   - Added `useModal.ts` to centralize modal state management across components.

5. **Loader Integration**:
   - Integrated the `Loader.ts` component into the following files:
     - `confirmEmail.tsx`
     - `expiredEmail.tsx`
     - `profile.tsx`
     - `settings.tsx`
     - `generalInfo.tsx`

6. **Subscription Utilities**:
   - Created a new utility file `subscriptionUtils.ts` for managing subscription-related logic.
   - Added `getBaseUrl.ts` to encapsulate base URL retrieval logic.

### Updates
1. **Profile Loading Check**:
   - Enhanced loading state checks in `profile.tsx` to improve user experience during data fetching.

2. **Account Management Logic**:
   - Updated `useAccountManagement.ts` to include logic for handling auto-renewal and displaying the next payment date.
   - Added comments for clarity and maintainability.

3. **Sidebar Refactoring**:
   - Refactored `sidebar.tsx` and `useSidebar.ts` to improve readability and performance.
   - Added comments throughout the code for better understanding of logic, including handling of end dates and next payment dates.

4. **Global Styles Update**:
   - Updated `_global.scss` to set margin to 0 for `<h2>` elements for consistent styling across the application.

5. **Confirmation Modal Enhancements**:
   - Updated the confirmation modal to include customizable button titles for improved user interaction.

6. **Sign-In Updates**:
   - Updated `signIn.tsx` and `useSignIn.ts` for better handling of sign-in logic and state management.

### Refactoring
- Refactored existing components and hooks to improve structure, readability, and maintainability.
- Ensured that all new features are well-documented with comments explaining the purpose and functionality of each piece of code.

### Conclusion
This PR enhances the overall functionality of account management, improves user experience with loading indicators, and refines existing components through thoughtful refactoring. The addition of utility functions and hooks aims to promote code reuse and maintainability throughout the application.

### Screenshot
Screenshot_1

### CODE REVIEW

File: src/features/profile/ui/settings/accountManagement/useAccountManagement.ts [hunk 1]
```diff
@@ -75,28 +97,30 @@ export const useAccountManagement = () => {
...
-
   //get end date of subscription and next payment date
   const {
     endDate: endDateOfSubscription,
     isSubscriptionActive,
     nextPaymentDate,
   } = computeSubscriptionDates(currentPayments)
 
+  const [accountType, setAccountType] = useState<AccountType>(
+    isSubscriptionActive ? 'business' : 'personal'
```
Comment (EvgeniyaYakunina) [inline src/features/profile/ui/settings/accountManagement/useAccountManagement.ts:108]: Предлагаю тип подписки вынести в константы, так как они используются не единожды.

File: src/features/profile/ui/settings/accountManagement/useAccountManagement.ts [hunk 2]
```diff
@@ -54,7 +69,7 @@ export const useAccountManagement = () => {
     defaultValues: {
       amount: 10,
       autoRenewal: false,
-      baseUrl: BASE_URL,
+      baseUrl: getBaseUrl(),
       paymentType: 'STRIPE',
       typeSubscription: 'DAY',
```
Comment (EvgeniyaYakunina) [inline src/features/profile/ui/settings/accountManagement/useAccountManagement.ts:95]: Предлагаю продолжительность подписки и тип оплаты вынести в константы.

File: src/features/profile/ui/settings/accountManagement/useAccountManagement.ts [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
 import { ErrorResponse } from '@/shared/api'
 import { PAYMENT_DELAY } from '@/shared/config'
 import { checkErrorMessages, useDelayedLoading } from '@/shared/utils'
 import { zodResolver } from '@hookform/resolvers/zod'
 import { useRouter } from 'next/router'
 import { z } from 'zod'
 
-const BASE_URL = process.env.NEXT_PUBLIC_BASE_URL || process.env.NEXT_PUBLIC_BASE_URL_VERCEL || ''
-
 type AccountType = 'business' | 'personal'
```
Comment (EvgeniyaYakunina) [inline src/features/profile/ui/settings/accountManagement/useAccountManagement.ts:24]: Мне кажется этот тип можно вынести в types, так как он является частью типов из апишки.

File: src/features/profile/ui/settings/accountManagement/accountManagements.tsx [hunk 4]
```diff
@@ -61,36 +62,31 @@ export const AccountManagements = () => {
   if (showLoading || isFetchingProfile) {
     return <Loader />
   }
+  const buttonTitleModal = modalTitle === 'Success' ? 'Ok' : 'Back to payment'
+  const contentModal =
+    modalTitle === 'Success'
+      ? 'Payment was successful'
+      : 'Transaction failed.Please,write to support'
```
Comment (EvgeniyaYakunina) [inline src/features/profile/ui/settings/accountManagement/accountManagements.tsx:71]: нужна TODO чтобы не забыть тут useTranslate сделать

**Review by Emilo067 — APPROVED** [review]

**Comment by EvgeniyaYakunina** [comment]
run build production

**Pull request merged**